cli/lib/kontena/cli/nodes/update_command.rb

```diff
@@ -3,16 +3,35 @@ class UpdateCommand < Kontena::Command
     include Kontena::Cli::Common
     include Kontena::Cli::GridOptions
 
+    requires_current_master
+    requires_current_master_token
+    requires_current_grid
+
     parameter "NODE", "Node name"
+
     option ["-l", "--label"], "LABEL", "Node label", multivalued: true
+    option "--clear-labels", :flag, "Clear node labels"
+    option ["--token"], "TOKEN", "Node token"
+    option "--generate-token", :flag, "Generate new node token"
+
+    def update_token
+      data = {}
+
+      data[:token] = self.token if self.token
+
+      spinner "Updating node #{self.node.colorize(:cyan)} token" do
+        client.put("nodes/#{current_grid}/#{self.node}/token", data)
+      end
+    end
 
     def execute
-      require_api_url
-      require_current_grid
-      token = require_token
+      self.update_token if self.token or self.generate_token?
```

Comment:
Cli should prompt and warn user here.. once user has accepted side-effect and token is updated cli should say some guide how to proceed to fix connection to master.